int32 id
int32 nChannels 
int16 params
channel[] channels